posts/2023-09-26-23.0.0.10-beta.adoc

```diff
@@ -0,0 +1,635 @@
+
+Primary Keys are LTPA keys in the specified keys default `ltpa.keys` file. Primary keys are used both for generating new LTPA tokens and for validating LTPA tokens. There can only be one primary keys file per Liberty Server.
+
+Validation keys are LTPA keys in any other `.keys` files other than the primary keys file. The validation keys are only used for validating LTPA tokens. They are _not_ used for generating new LTPA tokens. All validation keys must be located in the same directory as the primary keys file.
+
+  
+There are 2 ways to enable LTPA keys rotation without a planned outage:
+
+*  Use the Liberty Server's new capability to monitor the directory of the primary keys file for any new validation keys files. Enable the `monitorDirectory` and `monitorInterval` attributes. For example, add the following configurations to the `server.xml`:
++
+[source,xml]
+----
+<ltpa monitorDirectory="true" monitorInterval="5m"/>
+
+----
++
+The `monitorDirectory` attribute monitors the `${server.config.dir}/resources/security/` directory by default, but can monitor any directory the primary keys file is specified in. The directory monitor looks for any LTPA keys files with the `.keys` extension. The Open Liberty server reads these LTPA keys and uses them as Validation Keys.
++  
+If the `monitorInterval` is set to `0`, the default value, the directory stops being monitored.
++
+The `ltpa.keys` file can be renamed, for example, `validation1.keys` and then the Liberty server automatically regenerates a new `ltpa.keys` file with new primary keys that are used for all new LTPA tokens created. The keys in `validation1.keys` continue to be used for validating existing LTPA tokens. 
++   
+When the `validation1.keys` are no longer needed, remove them from usage by deleting the file, or by setting `monitorDirectory` to `false`. It is recommended to remove validation keys as it can improve performance.
+
+* Specify the validation keys file and optionally specify a date-time to stop using the validation keys.
+   1. Copy the primary keys file (`ltpa.keys`) to a validation keys file, for example `validation1.keys`.
+   2. Modify the server configuration to use the validation keys file  by specifying a `<validationKeys>` server configuration element inside the `<ltpa>` element. For example, add the following configuration to the `server.xml` file:
++
+[source,xml]
+----
+<ltpa>
+    <validationKeys fileName="validation1.keys" password="{xor}Lz4sLCgwLTs=" notUseAfterDate="2024-01-02T12:30:00Z"/>
+<ltpa/>  
+----
++
+The `validation1.keys` file can be removed from usage at a specified date-time in the future with the optional attribute, `notUseAfterDate`.  It is recommended to use `notUseAfterDate` to remove validation keys after a given period as it can improve performance.

```

Comment:
```suggestion
The `validation1.keys` file can be removed from usage at a specified date-time in the future with the optional attribute, `notUseAfterDate`.  It is recommended to use `notUseAfterDate` to ignore validation keys after a given period as it can improve performance.

```